synapse/federation/federation_server.py

```diff
@@ -632,7 +634,7 @@ async def on_send_leave_request(
 
     async def on_make_knock_request(
         self, origin: str, room_id: str, user_id: str, supported_versions: List[str]
-    ) -> Dict[str, Union[EventBase, str]]:
+    ) -> Dict[str, Union[JsonDict, str]]:
```

Comment:
Maybe this is simpler?

```suggestion
    ) -> JsonDict:
```